Realm/Tests/UtilTests.mm

```diff
@@ -56,6 +56,52 @@ - (void)testRLMExceptionWithCPlusPlusException {
                                      [NSException exceptionWithName:RLMExceptionName reason:@"Reason" userInfo:expectedUserInfo]));
 }
 
+- (void)testSystemExceptionWithPOSIXSystemException {
+    int code = ENOENT;
+    NSString *description = @"No such file or directory";
+
+    std::system_error exception(code, std::generic_category());
+    NSDictionary *expectedUserInfo = @{
+                                       NSLocalizedDescriptionKey : description,
+                                       @"Error Code" : @(code),
+                                       @"Category": [NSString stringWithUTF8String:std::generic_category().name()]
+                                       };
+    XCTAssertEqualObjects(RLMMakeError(exception),
+                          [NSError errorWithDomain:NSPOSIXErrorDomain code:code userInfo:expectedUserInfo]);
+}
+
+- (void)testSystemExceptionWithNonPOSIXSystemException {
+    int code = 999;
+    NSString *description = @"unspecified system_category error";
+
+    std::system_error exception(code, std::system_category());
+    NSDictionary *expectedUserInfo = @{
+                                       NSLocalizedDescriptionKey : description,
+                                       @"Error Code" : @(code),
+                                       @"Category": [NSString stringWithUTF8String:std::system_category().name()]
+                                       };
+    XCTAssertEqualObjects(RLMMakeError(exception),
+                          [NSError errorWithDomain:RLMUnknownSystemErrorDomain code:code userInfo:expectedUserInfo]);
+}
+
+using namespace realm;
+#import "shared_realm.hpp"
```

Comment:
Move to the top of the file after the other imports, and only use `using namespace` after the last header inclusion, as otherwise it'll apply to those headers and possibly break things.